backend/services/esweService.ts

```diff
@@ -105,72 +126,82 @@ export default class EsweService {
-    const mathsGrade = getLatestGrade(learnerLatestAssessments, AssessmentQualificationType.Maths)
-    const digitalLiteracyGrade = getLatestGrade(learnerLatestAssessments, AssessmentQualificationType.DigitalLiteracy)
+      const createSkillAssessmentSummary = (learnerAssessment: curious.LearnerAssessment) => {
+        const { qualification, establishmentName } = learnerAssessment || {}
+        const { qualificationType, qualificationGrade, assessmentDate } = qualification || {}
 
-    const createSkillAssessmentSummary = (learnerAssessment: curious.LearnerAssessment) => {
-      const { qualification, establishmentName } = learnerAssessment || {}
-      const { qualificationType, qualificationGrade, assessmentDate } = qualification || {}
+        if (!assessmentDate) {
+          return [
+            {
+              label: qualificationType === AssessmentQualificationType.English ? 'English/Welsh' : qualificationType,
+              value: 'Awaiting assessment',
+            },
+          ]
+        }
 
-      if (!assessmentDate) {
         return [
           {
-            label: qualificationType,
-            value: 'Awaiting assessment',
+            label: qualificationType === AssessmentQualificationType.English ? 'English/Welsh' : qualificationType,
+            value: qualificationGrade,
+          },
+          {
+            label: 'Assessment date',
+            value: readableDateFormat(assessmentDate, CURIOUS_DATE_FORMAT),
+          },
+          {
+            label: 'Assessment location',
+            value: establishmentName,
           },
         ]
       }
 
-      return [
-        {
-          label: qualificationType === AssessmentQualificationType.English ? 'English/Welsh' : qualificationType,
-          value: qualificationGrade,
-        },
-        {
-          label: 'Assessment date',
-          value: readableDateFormat(assessmentDate, CURIOUS_DATE_FORMAT),
-        },
-        {
-          label: 'Assessment location',
-          value: establishmentName,
-        },
-      ]
-    }
-
-    const functionalSkillLevels = {
-      english: createSkillAssessmentSummary(englishGrade),
-      maths: createSkillAssessmentSummary(mathsGrade),
-      digiLit: createSkillAssessmentSummary(digitalLiteracyGrade),
+      skillLevels = {
+        english: createSkillAssessmentSummary(englishGrade),
+        maths: createSkillAssessmentSummary(mathsGrade),
+        digiLit: createSkillAssessmentSummary(digitalLiteracyGrade),
+      }
+    } catch (e) {
+      if (e.response?.body?.errorCode === 'VC500') {
+        log.info(`Offender record not found in Curious.`)
+        return createFlaggedContent(DEFAULT_SKILL_LEVELS)
+      }
+      log.error(`Failed to get latest learning assessments. Reason: ${e.message}`)
     }
-
-    return createFlaggedContent(functionalSkillLevels)
+    return createFlaggedContent(skillLevels)
```

Comment:
Looking good @natclamp-moj . I think this function has gained the 'huge' status though.

Perhaps it could be broken down a bit to improve readability and help with maintenance. What do you think?